src/Microsoft.AspNetCore.Razor.Language/Legacy/TokenizerBackedParser.cs

```diff
@@ -228,13 +224,13 @@ protected internal bool Balance(BalancingModes mode)
             return Balance(mode, left, right, start);
         }
 
-        protected internal bool Balance(BalancingModes mode, TTokenType left, TTokenType right, SourceLocation start)
+        protected internal bool Balance(BalancingModes mode, SyntaxKind left, SyntaxKind right, SourceLocation start)
         {
             var startPosition = CurrentStart.AbsoluteIndex;
             var nesting = 1;
             if (!EndOfFile)
             {
-                var syms = new List<TToken>();
+                var syms = new List<SyntaxToken>();
```

Comment:
tokens, looks like a few places below too. I'd just do a find and replace `sym` => `token` and `syms` => `tokens` in this doc